File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.py
class Solution:
    def reverseWords(self, s: str) -> str:
        ans, temp = "", ""
        for i in s:
            if i == " ":
                ans += temp[::-1]
                ans += " "
                temp = ""
            else:
                temp += i
        
        return ans + temp[::-1]

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.swift
class Solution {
    func reverseWords(_ s: String) -> String {
        var ans = ""
        var temp = ""

        for i in s {
            if i == " " {
                ans += String(temp.reversed())
                ans += " "
                temp = ""
            } else {
                temp += String(i)
            }
        }

        return ans + String(temp.reversed())
    }
}